src/main/java/com/example/demo/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,26 @@
+package com.example.demo.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+    private String message;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstFieldName = constraintAnnotation.first();
+        secondFieldName = constraintAnnotation.second();
+        message = constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+        Object field = new BeanWrapperImpl(object).getPropertyValue(this.firstFieldName);
+        Object fieldMatch = new BeanWrapperImpl(object).getPropertyValue(this.secondFieldName);
```

Comment:
`this` is redundant here
```suggestion
        Object field = new BeanWrapperImpl(object).getPropertyValue(firstFieldName);
        Object fieldMatch = new BeanWrapperImpl(object).getPropertyValue(secondFieldName);
```